### PR TITLE
[front] - fix(website): website headers edition

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR fixes an issue with the website headers edition in the `VaultWebsiteModal` component. 

The main changes include:
- Extracting the AdvancedSettingsModal into a separate component.
- Fix the header mapping in the `WebCrawlerConfigurationResource` to correctly associate headers with their respective connectors.

**References:**
- https://github.com/dust-tt/tasks/issues/1532

## Risk

Low

## Deploy Plan

- Deploy `connectors`
- Deploy `front`

### CODE REVIEW

File: connectors/src/resources/webcrawler_resource.ts [hunk 1]
```diff
@@ -90,19 +90,30 @@ export class WebCrawlerConfigurationResource extends BaseResource<WebCrawlerConf
...
-      if (r) {
-        r.headers[header.key] = header.value;
-      }
+    const configurationHeaders = await WebCrawlerConfigurationHeader.findAll({
+      where: {
+        webcrawlerConfigurationId: blobs.map((b) => b.id),
+      },
     });
 
+    const configToConnectorMap = blobs.reduce(
```
Comment (Fraggle) [inline connectors/src/resources/webcrawler_resource.ts:99]: nit: `configToConnectorId`

**Review by Fraggle — APPROVED** [review]

File: front/components/vaults/VaultWebsiteModal.tsx [hunk 2]
```diff
@@ -674,3 +610,97 @@ export default function VaultWebsiteModal({
     </Modal>
   );
 }
+
+interface AdvancedSettingsModalProps {
+  advancedSettingsOpened: boolean;
+  setAdvancedSettingsOpened: (open: boolean) => void;
+  headers: { key: string; value: string }[];
+  setHeaders: (headers: { key: string; value: string }[]) => void;
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/components/vaults/VaultWebsiteModal.tsx:618]: Simply for the sake of readability, that's why I kept it in the same file

[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/components/vaults/VaultWebsiteModal.tsx:618]: Do you feel it should live within the `VaultWebsiteModal`?

[the remark this answers is not in the document]
Comment (Fraggle) [reply front/components/vaults/VaultWebsiteModal.tsx:618]: I don't have a strong opinion :)
I would have said nothing if the interface of `AdvancedSettingsModal` would have been something like that.

```ts
{
 isOpen: boolean;
 close: () => void;
 initialSettings: { key: string; value: string }[];
 onSave: ({ key: string; value: string }[]) => void;
}
```

It's similar but the settings state is handled directly within the component.

[the remark this answers is not in the document]
Comment (JulesBelveze) [reply front/components/vaults/VaultWebsiteModal.tsx:618]: You right 👍🏼

**Pull request merged**